documentation/modules/ROOT/pages/metrics-alerting.adoc

```diff
@@ -0,0 +1,342 @@
+alerts are defined using a `PrometheusRule` object. The following YAML
+defines a `PrometheusRule` that will cause an alert to fire when the number
+of `404` errors in the `http_requests_total` exceeds a quantity of 10:
+
+[.console-input]
+[source,yaml]
+----
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: example-alert
+spec:
+  groups:
+  - name: example
+    rules:
+    - alert: TooManyErrorAlert
+      expr: http_requests_total{code="404"} > 10
+----
+
+1. Make sure you are in the _Developer_ perspective
+2. Click _+Add_
+3. Click _YAML_
+4. Copy and paste the above `PrometheusRule` YAML into the box
+5. Click _Create_
+
+[WARNING]
+====
+With OpenShift 4.5, alerts are not yet exposed in the OpenShift web console,
+even for administrators. Custom monitoring and alerting is a Tech Preview
+feature. In the future, the alerts will move into the web console for both
+admin and non-admin users. The following instructions are temporary until
+OpenShift 4.6 is available.
+====
+
+### Triggering the Alert
+Triggering the alert is simple. Visit the `/err` endpoint of the application
+until you have exceeded more than 10 `404` codes. You can check how many
+`404` you have either by viewing the `/metrics` endpoint, or by using the
+metrics view in the OpenShift web console!
+
+Once you have more than 10 `404` errors:
+
+1. Switch to the _Administrator_ perspective at the upper left
+2. Click _Monitoring_
+3. Click _Alerting_
+4. Click _Alertmanager UI_
+5. Click _Log in with OpenShift_
+6. Provide your cluster admin credentials
+7. Click _Allow selected permissions_
+
+You should see your alert listed.
+
+[NOTE]
+====
+The above process for accessing Alertmanager is because OpenShift places an
+OAuth proxy in front. You are logging in (via the OAuth proxy) and then
+granting permission to use your user credentials (via the proxy).
+Alertmanager itself does not have any authentication, so placing it behind
+the proxy and requiring `cluster-admin` credentials ensures that only the
+right people can see it.
+====
+
+### Recording Rules
+Earlier we mentioned that it was possible to perform mathematical calculations on recorded metrics. This is done via a _Recording Rule_, which is a component of a `PrometheusRule`. We won't do more than show a potential example of a rule here:
+
+[source,yaml]
+----
+  - name: example
+    rules:
+    - record: job:http_inprogress_requests:sum
+      expr: sum by (job) (http_inprogress_requests)
+----
+
+This would calculate a sum of all `job` values in the `http_inprogress_requests` key. 
+
+It is important for you and your developers to understand that _Recording
+Rules_ can be expensive in terms of the calculation power they require.
+Complex calculation expressions will consume Prometheus' horsepower, and it
+is possible to cripple the monitoring infrastructure by writing too many
+expensive/complicated recording rule expressions. Keep it simple.
```

Comment:
Any closing remarks/next steps to add?

Reply:
added a paragraph at the end for _Next Steps_ with a link to PromQL docs.